include/gauge_field.h

```diff
@@ -393,6 +393,13 @@ namespace quda {
      */
     virtual void copy(const GaugeField &src) = 0;
 
+    /**
+     * @brief Generic gauge field shift
+     * @param[in] src Source from which we are shifting (extended field in case of MPI)
+     * @param[in] dx Host array of shifts to apply to the field
+     */
+    virtual void shift(const GaugeField &src, const int *dx) = 0;
```

Comment:
I think we should use the  QUDA `array` class here for `dx`, rather than passing `const int *`.  This is a change I've started to propagate in other parts already.

Reply:
[a304a3a](https://github.com/lattice/quda/pull/1348/commits/a304a3ab0225e254f0afb7eb1be7475f39a9bd6b)